File: lib_db_news/src/main/java/com/fly/tour/db/NewsDBConfig.java
package com.fly.tour.db;

/**
 * Description: <NewsDBConfig><br>
 * Author:      gxl<br>
 * Date:        2019/5/24<br>
 * Version:     V1.0.0<br>
 * Update:     <br>
 */
public interface NewsDBConfig {
    String DB_NAME = "fly_news.db";
    int VERSION_CODE = 1;


    interface NewsType {
        String TABLE_NAME = "news_type";
        String CLUMN_TYPE_NAME = "typename";
        String CREATE_TABLE = "create table " + TABLE_NAME + " (id integer PRIMARY KEY autoincrement," + CLUMN_TYPE_NAME + " varchar(64))";
    }

    interface NewsDetail {
        String TABLE_NAME = "news_detail";
        String CLUMN_TYPE_ID = "typeid";
        String CLUMN_TITLE = "title";
        String CLUMN_CONTENT = "content";
        String CREATE_TABLE = "create table " + TABLE_NAME + " (id integer PRIMARY KEY autoincrement, " + CLUMN_TYPE_ID + " integer," + CLUMN_TITLE + " varchar(64)," + CLUMN_CONTENT + " text)";

    }
}

File: lib_fly_db/src/main/java/com/fly/tour/db/NewsDBConfig.kt
package com.fly.tour.db

/**
 * Description: <NewsDBConfig><br>
 * Author:      mxdl<br>
 * Date:        2019/5/24<br>
 * Version:     V1.0.0<br>
 * Update:     <br>
 */
interface NewsDBConfig {

    interface NewsType {
        companion object {
            val TABLE_NAME = "news_type"
            val CLUMN_ID = "id"
            val CLUMN_TYPE_NAME = "typename"
            val CLUMN_ADD_TIME = "addtime"
            val CREATE_TABLE = "create table $TABLE_NAME ($CLUMN_ID integer PRIMARY KEY autoincrement,$CLUMN_TYPE_NAME varchar(64),$CLUMN_ADD_TIME datetime default (datetime('now','localtime')))"
        }
    }

    interface NewsDetail {
        companion object {
            val TABLE_NAME = "news_detail"
            val CLUMN_ID = "id"
            val CLUMN_TYPE_ID = "typeid"
            val CLUMN_TITLE = "title"
            val CLUMN_CONTENT = "content"
            val CLUMN_ADD_TIME = "addtime"
            val CREATE_TABLE = "create table $TABLE_NAME ($CLUMN_ID integer PRIMARY KEY autoincrement, $CLUMN_TYPE_ID integer,$CLUMN_TITLE varchar(64),$CLUMN_CONTENT text,$CLUMN_ADD_TIME datetime default (datetime('now','localtime')))"
        }
    }

    companion object {
        val DB_NAME = "fly_news.db"
        val VERSION_CODE = 1
    }
}
